### ISSUE TITLE
RFC search_index and search_type configuration

### ISSUE THREAD

**Issue by hachreak** [inveniosoftware/invenio-records-rest#102]
Can we simplify the `RECORDS_REST_ENDPOINTS` configuration removing the `search_index` and `search_type` configuration and use directly the configuration in `search_class`?

[Here](https://github.com/inveniosoftware/invenio-records-rest/blob/master/invenio_records_rest/views.py#L189-L201) you can see how it works now.
If you have 2 points where you can configure them, it can confuse the user and induce possible misconfiguration.
E.g. the default Search Class in invenio-search follows its [own logic](https://github.com/inveniosoftware/invenio-search/blob/master/invenio_search/api.py#L72-L73) to decide which it will use, that can be different from the logic applied here.

I think it can be simplified with something like:

```python
search_index = search_class.Meta.index
search_type = search_class.Meta.doc_types
```

----------------------------------------------------------------------
cc @jirikuncar @tiborsimko @lnielsen 

See https://github.com/inveniosoftware/invenio-records-rest/issues/99

**Comment by jirikuncar** [comment]
@hachreak you can use same search class with different index or doc_type.

**Comment by hachreak** [comment]
In a Object Oriented context I think it makes sense put all the configuration inside the Object.
In case you need another index configuration you can easily extend the class and configure it differently.

**Comment by jirikuncar** [comment]
@hachreak you can override the `Meta` object inside the class but one needs to write a code. Having configuration allows instances with less developers simply configure it without subclassing. We should make sure that it's well explained in documentation.

**Comment by hachreak** [comment]
do we still require more documentation than we already have?

**Comment by jirikuncar** [comment]
Closed via b4f90a2
